### PR TITLE
etcd-tester: return error for mismatch rev/hash

### PR DESCRIPTION
**Pull request by gyuho**
@xiang90 Can't remember what was wrong with printing? 

Is is other than printing the wrong number of compacting entries (already fixed)?

### CODE REVIEW

File: tools/functional-tester/etcd-tester/tester.go [hunk 1]
```diff
@@ -174,21 +170,25 @@ func (tt *tester) checkConsistency() (failed bool, err error) {
 	}
 	plog.Printf("%s updated current revisions with %d", tt.logPrefix(), tt.currentRevision)
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: this line should not be printed if the hash are not consistent. also if the hash is not consistent, there is no point to keep revision. the cluster will be cleaned up anyway.

**Comment by gyuho** [comment]
@xiang90 Right. Addressed. PTAL. Thanks.

File: tools/functional-tester/etcd-tester/tester.go [hunk 2]
```diff
@@ -172,23 +168,26 @@ func (tt *tester) checkConsistency() (failed bool, err error) {
 
 		plog.Printf("%s #%d inconsistent current revisions %+v", tt.logPrefix(), i, revs)
 	}
-	plog.Printf("%s updated current revisions with %d", tt.logPrefix(), tt.currentRevision)
-
-	if !ok || err != nil {
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: can we keep the previous log? just change failed = true to return error?

File: tools/functional-tester/etcd-tester/tester.go [hunk 3]
```diff
@@ -172,23 +168,26 @@ func (tt *tester) checkConsistency() (failed bool, err error) {
...
-		plog.Printf("%s checking current storage hashes failed [hashes: %v]", tt.logPrefix(), hashes)
-		failed = true
+		err = fmt.Errorf("inconsistent hashes [%v]", hashes)
+		plog.Printf("%s %v", tt.logPrefix(), err)
 		return
 	}
 	plog.Printf("%s all members are consistent with storage hashes", tt.logPrefix())
-	return
+
+	plog.Printf("%s updated current revisions with %d", tt.logPrefix(), tt.currentRevision)
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/tester.go]: update current revision to %d

**Comment by gyuho** [comment]
Just addressed. PTAL. Thanks.

**Comment by xiang90** [comment]
lgtm

**Pull request merged**